tests/integration-tests/src/test/java/org/apache/activemq/artemis/tests/integration/paging/PagingSendTest.java

```diff
@@ -288,6 +296,408 @@ public void testPagingDoesNotDuplicateBatchMessagesAfterPagingStarted() throws E
+         addressSettings.setPageFullMessagePolicy(PageFullMessagePolicy.FAIL);
+         addressSettings.setPageSizeBytes(size);
+         addressSettings.setPageLimitBytes(Long.valueOf(size * 5));
+         addressSettings.setMaxSizeBytes(size);
+         addressSettings.setPageLimitMessages(maxMessages);
+
+         server.getAddressSettingsRepository().addMatch(queueName, addressSettings);
+
+         int totalMessages = 0;
+         int messageSize = 1024;
+         ClientProducer producer = session.createProducer(queueAddr);
+         boolean stop = false;
+         while (!stop) {
+            try {
+               ClientMessage message = createMessage(session, messageSize, totalMessages, null);
+               producer.send(message);
+               totalMessages++;
+               session.commit();
+            } catch (ActiveMQAddressFullException ex) {
+               stop = true;
+            }
+         }
+
+         Queue queue = server.locateQueue(queueAddr);
+         assertTrue(waitForMessages(queue, totalMessages, 10000));
+
+         PagingStore queuePagingStore = queue.getPagingStore();
+         assertTrue(queuePagingStore != null && queuePagingStore.isPageFull());
+
+         // the pages reach the limit
+         long existingPages = queuePagingStore.getNumberOfPages();
+         long maxPages = queuePagingStore.getPageLimitBytes() / queuePagingStore.getPageSizeBytes();
+
+         assertTrue(existingPages > maxPages);
+
+         // but messages still under limit
+         PageCursorProvider cursorProvider = queuePagingStore.getCursorProvider();
+         Long existingMessages = PageCursorProviderTestAccessor.getNumberOfMessagesOnSubscriptions(cursorProvider);
+         assertTrue(existingMessages < maxMessages, "existing " + existingMessages + " should be less than max " + maxMessages);
+
+         server.stop(true);
+         waitForServerToStop(server);
+
+         // restart the server the pageFull is still true
+         try {
+            server.start();
+            waitForServerToStart(server);
+
+            queue = server.locateQueue(queueAddr);
+
+            queuePagingStore = queue.getPagingStore();
+            assertTrue(queuePagingStore != null && queuePagingStore.isPageFull());
+
+            // but messages still under limit
+            cursorProvider = queuePagingStore.getCursorProvider();
+            existingMessages = PageCursorProviderTestAccessor.getNumberOfMessagesOnSubscriptions(cursorProvider);
+            assertTrue(existingMessages < maxMessages, "existing " + existingMessages + " should be still less than max " + maxMessages);
+
+            // now increase the pageLimitBytes to unblock procuder
+            addressSettings.setPageLimitBytes(Long.valueOf(size * 20));
+            server.getAddressSettingsRepository().addMatch(queueName, addressSettings);
+
+            // no longer page full
+            assertTrue(queuePagingStore != null && !queuePagingStore.isPageFull());
```

Comment:
queuePagingStore != null is superfluous her, since it already checked it wasnt null, and havent changed the variable since (but have used it again without checking it isnt null)